### PR TITLE
Improve the doc for the api.

### PR DESCRIPTION
**Pull request by Chouby**
Everything is in the title.

### CODE REVIEW

File: include/api.php [hunk 1]
```diff
@@ -38,14 +37,14 @@ function pll_the_languages( $args = array() ) {
...
- * Returns the language set in admin language filter on backend ( false if set to all languages )
+ * Returns the current language on frontend.
+ * Returns the language set in admin language filter on backend ( false if set to all languages ).
  *
  * @api
  * @since 0.8.1
  *
- * @param string $field Optional, the language field to return ( see PLL_Language ), defaults to 'slug', pass OBJECT constant to get the language object.
- * @return string|PLL_Language|bool The requested field for the current language
+ * @param string $field Optional, the language field to return ( @see PLL_Language ), defaults to 'slug'. Pass OBJECT constant to get the language object.
```
Comment (raaaahman) [inline include/api.php:46]: We should choose between using two of our own constants, or two strings.

File: include/api.php [hunk 2]
```diff
@@ -122,15 +121,17 @@ function pll_home_url( $lang = '' ) {
...
  * @since 0.6
  *
- * @param string $name      a unique name for the string
- * @param string $string    the string to register
- * @param string $context   optional the group in which the string is registered, defaults to 'polylang'
- * @param bool   $multiline optional whether the string table should display a multiline textarea or a single line input, defaults to single line
+ * @param string $name      A unique name for the string.
+ * @param string $string    The string to register.
+ * @param string $context   Optional, the group in which the string is registered, defaults to 'polylang'.
+ * @param bool   $multiline Optional, true if the string table should display a multiline textarea,
```
Comment (raaaahman) [inline include/api.php:132]: Don't we use only multiline textareas now?

File: include/api.php [hunk 3]
```diff
@@ -337,74 +344,88 @@ function pll_save_post_translations( $arr ) {
...
- * Returns an array of translations of a post
+ * Returns an array of translations of a post.
  *
  * @api
  * @since 1.8
  *
- * @param int $post_id
- * @return array an associative array of translations with language code as key and translation post_id as value
+ * @param int $post_id Post id.
+ * @return int[] An associative array of translations with language code as key and translation post id as value.
```
Comment (raaaahman) [inline include/api.php:389]: So it is not really an `int[]`, is it?

File: include/api.php [hunk 4]
```diff
@@ -337,74 +344,88 @@ function pll_save_post_translations( $arr ) {
...
- * Returns an array of translations of a term
+ * Returns an array of translations of a term.
  *
  * @api
  * @since 1.8
  *
- * @param int $term_id
- * @return array an associative array of translations with language code as key and translation term_id as value
+ * @param int $term_id Term id.
+ * @return int[] An associative array of translations with language code as key and translation term id as value.
```
Comment (raaaahman) [inline include/api.php:402]: Same as above.

**Review by raaaahman — COMMENTED** [review]
What does 'language code' refers to?

- A W3C locale?
- A WordPress locale?
- A slug?

File: include/api.php (hunk 3 above)
Comment (Chouby) [reply include/api.php:389]: Values are only integers. This is waht the static analysis checks wehne accessing a value.

File: include/api.php (hunk 2 above)
Comment (Chouby) [reply include/api.php:132]: In `PLL_Translated_Option` yes. But this function still offers the possibility to use a single line input.

File: include/api.php (hunk 1 above)
Comment (Chouby) [reply include/api.php:46]: Sorry, I don't understand.

**Comment by Chouby** [comment]
> What does 'language code' refers to?
> 
> * A W3C locale?
> * A WordPress locale?
> * A slug?

The "slug" is the word used by WordPress for taxonomy terms. I often used "language code" as used in the interface allowing to create or edit the languages.

File: include/api.php [hunk 5]
```diff
@@ -77,40 +76,40 @@ function pll_default_language( $field = 'slug' ) {
...
+ * Among the post and its translations, returns the id of the post which is in the language represented by $slug.
  *
  * @api
  * @since 0.5
  *
- * @param int    $post_id post id
- * @param string $slug    optional language code, defaults to current language
- * @return int|false|null post id of the translation if exists, false otherwise, null if the current language is not defined yet
+ * @param int    $post_id Post id.
+ * @param string $slug    Optional language code, defaults to the current language.
```
Comment (raaaahman) [inline include/api.php:85]: `$slug` might get confused with the post's slug.

File: include/api.php [hunk 6]
```diff
@@ -77,40 +76,40 @@ function pll_default_language( $field = 'slug' ) {
...
+ * Among the term and its translations, returns the id of the term which is in the language represented by $slug.
  *
  * @api
  * @since 0.5
  *
- * @param int    $term_id term id
- * @param string $slug    optional language code, defaults to current language
- * @return int|false|null term id of the translation if exists, false otherwise, null if the current language is not defined yet
+ * @param int    $term_id Term id.
+ * @param string $slug    Optional language code, defaults to the current language.
```
Comment (raaaahman) [inline include/api.php:99]: `$slug` might get confused with the term's slug.

**Review by raaaahman — CHANGES_REQUESTED** [review]
> The "slug" is the word used by WordPress for taxonomy terms. I often used "language code" as used in the interface allowing to create or edit the languages.

Okay if its the same terminology than the user interface.

File: include/api.php (hunk 5 above)
Comment (Chouby) [reply include/api.php]: This is in fact the language slug, used in the url, but you are right on the risk of confusion. I changed this.

**Review by raaaahman — APPROVED** [review]

**Pull request merged**